fetch/api/abort/general.js

```diff
@@ -0,0 +1,335 @@
+if (self.importScripts) {
+  // Load scripts if being run from a worker
+  importScripts(
+    '/resources/testharness.js',
+    '/common/utils.js'
+  );
+}
+
+// This is used to close connections that weren't correctly closed during the tests,
+// otherwise you can end up running out of HTTP connections.
+let requestKeys = [];
+
+function abortRequests() {
+  const keys = requestKeys;
+  requestKeys = [];
+  return Promise.all(
+    keys.map(key => fetch(`../resources/stash-put.py?key=${key}&value=close`))
+  );
+}
+
+function assert_abort_error(err) {
+  assert_equals(err.constructor, DOMException);
+  assert_equals(err.name, 'AbortError');
+}
+
+promise_test(async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  controller.abort();
+
+  await fetch('../resources/data.json', { signal }).then(() => {
+    assert_unreached("Fetch must not resolve");
+  }, err => {
+    // Using .catch rather than try/catch to ensure the promise
+    // is rejecting (rather than throwing).
+    assert_abort_error(err);
+  });
+}, "Aborting rejects with AbortError");
+
+test(() => {
+  // TODO: we may want to discuss this design idea
+  const request = new Request('');
+  assert_true(Boolean(request.signal), "Signal member is present & truthy");
+  assert_equals(request.signal.constructor, AbortSignal);
+}, "Request objects have a signal property");
+
+promise_test(async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  controller.abort();
+
+  const request = new Request('../resources/data.json', { signal });
+
+  // TODO: we may want to discuss this design idea
+  assert_true(Boolean(request.signal), "Signal member is present & truthy");
+  assert_equals(request.signal.constructor, AbortSignal);
```

Comment:
Though maybe `instanceof` is better so we don't have to update this tests and it'll keep passing in newer UAs.